editor/plugins/asset_library_editor_plugin.cpp

```diff
@@ -726,9 +726,9 @@ void EditorAssetLibrary::_image_update(bool use_cache, bool final, const PoolByt
 		uint8_t jpg_signature[3] = { 255, 216, 255 };
 
 		if (r.ptr()) {
-			if (memcmp(&r[0], &png_signature[0], 8) == 0) {
+			if ((memcmp(&r[0], &png_signature[0], 8) == 0) && Image::_png_mem_loader_func) {
 				image->copy_internals_from(Image::_png_mem_loader_func(r.ptr(), len));
-			} else if (memcmp(&r[0], &jpg_signature[0], 3) == 0) {
+			} else if ((memcmp(&r[0], &jpg_signature[0], 3) == 0) && Image::_jpg_mem_loader_func) {
```

Comment:
Actually, you don't need to parenthesize `==` here, since `&&` has lower priority.

Reply:
Yes. But I think there are many things going on in this condition expression. Parentheses can make this complex expression easier to read, so the reader won't have to remember the operator precedence thing.

ref:
* https://rules.sonarsource.com/c/RSPEC-864
* https://wiki.sei.cmu.edu/confluence/display/c/EXP00-C.+Use+parentheses+for+precedence+of+operation

Reply:
Well, you wouldn't parenthesize `a == 4 && c == 6`, would you? Either way, it is fine as-is.